### PR TITLE
[Impeller] disable impeller supports rendering if fuchsia.

### PR DESCRIPTION
**Pull request by jonahwilliams**
*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I signed the [CLA].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by dnfield — CHANGES_REQUESTED** [review]
I'd rather not do this.

These varaibles are defined above in terms of platforms, and we should never be in a position where `is_fuchsia` is true.

I'd rather add an assert somewhere in impeller.gni like `assert(!is_fuchsia)` for now.

**Comment by jonahwilliams** [comment]
if we're building on a linux host for fuchsia though, won't is_linux and is_fuchsia both be true? Genuinely don't know, trying it.

**Comment by jonahwilliams** [comment]
Moved the check up to the impeller_enable_opengles/impeller_enable_vulkan check and pulled in https://github.com/flutter/engine/pull/44005/files so we can verify if it works

File: impeller/tools/impeller.gni [hunk 1]
```diff
@@ -15,10 +15,10 @@ declare_args() {
   impeller_enable_metal = is_mac || is_ios
 
   # Whether the OpenGLES backend is enabled.
-  impeller_enable_opengles = is_linux || is_win || is_android
+  impeller_enable_opengles = (is_linux && !is_fuchsia) || is_win || is_android
```
[the remark this answers is not in the document]
Comment (zanderso) [reply impeller/tools/impeller.gni:18]: The `|| target_os == "fuchsia"` was probably a copy-paste from the Fuchsia tree. It doesn't look like it's needed in the Engine build.

**Comment by jonahwilliams** [comment]
This isn't sufficient, because these values aren't ever true at the same time like Dan says.

**Pull request closed without merging**